### PR TITLE
`Development`: Refactor result detail component

### PR DESCRIPTION
**Pull request by b-fein**
#### General

- [x] I tested **all** changes and their related features with **all** corresponding user types on a test server.
- [x] This is a small issue that I tested locally and was confirmed by another developer on a test server.
- [x] Language: I followed the [guidelines for inclusive, diversity-sensitive, and appreciative language](https://docs.artemis.ase.in.tum.de/dev/guidelines/language-guidelines/).
- [x] I chose a title conforming to the [naming conventions for pull requests](https://artemis-platform.readthedocs.io/en/latest/dev/guidelines/development-process.html#naming-conventions-for-github-pull-requests).
#### Client
- [x] I followed the [coding and design guidelines](https://docs.artemis.ase.in.tum.de/dev/guidelines/client/).
- [x] I added multiple integration tests (Jest) related to the features (with a high test coverage), while following the [test guidelines](https://docs.artemis.ase.in.tum.de/dev/guidelines/client-tests/).
- [x] I documented the TypeScript code using JSDoc style.

### Motivation and Context
Follow-up to #4835 where I noticed the pretty long methods in this component.

### Description
Makes the code more readable by splitting up methods. Adds additional unit tests for previously uncovered code elements.

Additionally lets the component handle cases where no exercise is given. I went through all usages of the component to pass an exercise wherever it is known so that the chart can still be shown in nearly all cases.

### Steps for Testing

Prerequisites:
- 1 Instructor
- Some programming exercise either already graded or not, optionally with static code analysis enabled.

1. Log in to Artemis
2. All feedback types (test case feedback, SCA feedback, manual feedback with/without grading instruction usage) should still be shown as usual in the dialogue when clicking on a result string (e.g., ‘12 of 12 passed, 100%’). No errors about a missing exercise should be printed in the console log.
   Those result strings are shown for example
   - in the online editor of programming exercises,
   - on the ‘Submissions’ page of an exercise accessible to instructors,
   - in the student’s view when looking at all exercises for a course,
   - in the student’s view when looking at an exercise they submitted to.

#### Code Review
- [x] Review 1
- [x] Review 2
#### Manual Tests
- [x] Test 1
- [x] Test 2

### Test Coverage

| Class/File | Branch | Line |
|------------|-------:|-----:|
| `result-detail.component.ts` | 87.7% | 94.4% |

### CODE REVIEW

File: src/main/webapp/app/exercises/shared/result/result-detail.component.ts [hunk 1]
```diff
@@ -336,11 +268,84 @@ export class ResultDetailComponent implements OnInit {
...
+        if (Feedback.isSubmissionPolicyFeedback(feedback)) {
+            return this.createProgrammingExerciseSubmissionPolicyFeedbackItem(feedback, previewText);
+        } else if (Feedback.isStaticCodeAnalysisFeedback(feedback)) {
+            return this.createProgrammingExerciseScaFeedbackItem(feedback);
+        } else if (feedback.type === FeedbackType.AUTOMATIC) {
+            return this.createProgrammingExerciseAutomaticFeedbackItem(feedback, previewText);
+        } else if ((feedback.type === FeedbackType.MANUAL || feedback.type === FeedbackType.MANUAL_UNREFERENCED) && feedback.gradingInstruction) {
+            return this.createProgrammingExerciseGradingInstructionFeedbackItem(feedback, previewText);
+        } else {
+            return {
```
Comment (Strohgelaender) [inline src/main/webapp/app/exercises/shared/result/result-detail.component.ts:288]: Maybe use another method for this as well? Just to have every case as a one-liner.

File: src/main/webapp/app/exercises/shared/result/result-detail.component.ts [hunk 2]
```diff
@@ -336,11 +268,84 @@ export class ResultDetailComponent implements OnInit {
...
+        const scaIssue = StaticCodeAnalysisIssue.fromFeedback(feedback);
+        const text = this.showTestDetails ? `${scaIssue.rule}: ${scaIssue.message}` : scaIssue.message;
+        const scaPreviewText = ResultDetailComponent.computeFeedbackPreviewText(text);
+
+        return {
+            type: FeedbackItemType.Issue,
+            category: 'Code Issue',
+            title: `${scaCategory} Issue in file ${ResultDetailComponent.getIssueLocation(scaIssue)}`.trim(),
+            text,
+            previewText: scaPreviewText,
```
Comment (Strohgelaender) [inline src/main/webapp/app/exercises/shared/result/result-detail.component.ts:337]:
```suggestion
        const previewText = ResultDetailComponent.computeFeedbackPreviewText(text);

        return {
            type: FeedbackItemType.Issue,
            category: 'Code Issue',
            title: `${scaCategory} Issue in file ${ResultDetailComponent.getIssueLocation(scaIssue)}`.trim(),
            text,
            previewText,
```

File: src/main/webapp/app/exercises/shared/result/result-detail.component.ts [hunk 3]
```diff
@@ -349,6 +354,53 @@ export class ResultDetailComponent implements OnInit {
...
+     * @private
+     */
+    private createProgrammingExerciseGradingInstructionFeedbackItem(feedback: Feedback, previewText?: string): FeedbackItem {
+        const gradingInstruction = feedback.gradingInstruction!;
+
+        return {
+            type: feedback.isSubsequent ? FeedbackItemType.Subsequent : FeedbackItemType.Feedback,
+            category: this.showTestDetails ? 'Tutor' : 'Feedback',
+            title: feedback.text,
+            text: feedback.detailText ? gradingInstruction.feedback + '\n' + feedback.detailText : gradingInstruction.feedback,
```
Comment (Strohgelaender) [inline src/main/webapp/app/exercises/shared/result/result-detail.component.ts:397]:
```suggestion
            text: gradingInstruction.feedback + (feedback.detailText ?  '\n' + feedback.detailText : ''),
```

**Review by Strohgelaender — APPROVED** [review]
Code looks good 👍

**Review by bassner — DISMISSED** [review]
LGTM!

**Review by bassner — APPROVED** [review]
Code looks good 👍 
Tested this on TS2 and saw no issues with the result detail component. No errors were logged and it works like intended

**Review by Dominik-Weinzierl — APPROVED** [review]
Code looks good to me and all cases work as expected :+1:

**Pull request merged**